mm/mm_gran/mm_gran.c

```diff
@@ -0,0 +1,138 @@
+/****************************************************************************
+ * mm/mm_gran/mm_gran.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <assert.h>
+#include <debug.h>
+
+#include <nuttx/mm/gran.h>
+
+#include "mm_gran/mm_gran.h"
+#include "mm_gran/mm_grantable.h"
+
+#ifdef CONFIG_GRAN
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+void gran_free(GRAN_HANDLE handle, FAR void *memory, size_t size)
```

Comment:
Can we keep the original file layout and make the change as small as possible? The change is unnecessary big and complex to remove the size limitation.